src/socket.cpp

```diff
@@ -3076,6 +2963,29 @@ bool sipMsgCheck (const char *P_msg, struct sipp_socket *socket)
     return false;
 }
 
+bool fill_sockaddr_from_ip(struct sockaddr_storage* saddr, char* ip, bool use_ipv6)
+{
+    struct addrinfo* h ;
+    struct addrinfo hints = {AI_PASSIVE, AF_UNSPEC,};
+    if (!getaddrinfo(ip, NULL, &hints, &h))
```

Comment:
~~~c
char service[std::numeric_limits<unsigned short>::digits10 + 1];
snprintf(service, sizeof(service), "%hs", local_port);
if (!getaddrinfo(ip, service, &hints, &h);
//...
~~~

And we don't have to worry about any of the problems now with `(_RCAST(struct sockaddr_in6 *, saddr))`. That said, I'd also consider this a temporary fix, ideally we should keep the port as a number.